qiime2/core/testing/plugin.py

```diff
@@ -135,7 +144,10 @@
     name='Concatenate integers',
     description='This method concatenates integers into a single sequence in '
                 'the order they are provided.',
-    citations=[citations['baerheim1994effect']]
+    citations=[citations['baerheim1994effect']],
+    examples={'concatenate_ints_simple': concatenate_ints_simple,
```

Comment:
Just curious, why have this as a `dict` rather than a `set` or `tuple`? It looks like the key names are the same as the value names, and (based on a *very* cursory static review) it doesn't seem like the keys relate back to (for example) descriptions as is the case with `parameters`?

Reply:
Hey @wasade!

It is so that you can pluck the example off of the action, like this:

```python
        action = self.plugin.actions['concatenate_ints']
        use = usage.DiagnosticUsage()
        action.examples['concatenate_ints_simple'](use)
```

Reply:
ahhhhh! :) thanks!

Reply:
We could populate a dict behind the scenes (similar to citations... ish)

Reply:
I'm good either way though